### PR TITLE
Add accent-insensitive search option

### LINKED ISSUE #6181: Search for result including accent characters
## Summary
Enable accent result in search result (possibly as a menu setting) after **searching for unaccent string** and vice versa.

## Examples
If there is "pouzivatel" written in keepass entry. Search for "používateľ" (so including accent characters) return no results. However, I can't guarantee that all users, who write into keepass file, will use the same approach and include or not accent characters. The same situation is if I search for "pouzivatel" but there is "používateľ" in entry.

## Context
If words with and without an accent are used both in the keepass database, the search is more complicated. It's the same with capital letters, but they work as expected 😉

### PR DESCRIPTION
**Pull request by yyyyyyyan**
Make search accent-insensitive by default: both the search term and
entry field values are normalized by decomposing to Unicode NFD and
stripping combining marks, so that e.g. "pouzivatel" matches "používateľ"
and vice versa.

The `+` (exact match) modifier implicitly forces accent-sensitive matching
per-term, giving users fine-grained control without a separate UI toggle.
For example, `+café` matches only "café" and not "cafe".

**AI Disclosure:** This PR was developed with assistance from Claude (Opus 4.6). I created an implementation plan in an interactive session with Claude, and Claude executed the initial implementation. After my review and tests I manually refined the code (added ASCII fast path, added some edge-case unit tests, fixed signal/multiplexer race condition in db open sequence).

Fixes #6181

## Screenshots
Accent-insensitive search (default): "testingabc" matches accented entry "téstîñgàbç":
image

Exact match (+) forces accent-sensitive: "+testingabc" does not match accented entry:
image

Exact match (+) with accented query: "+téstîñgàbç" matches the accented entry:
image

## Testing strategy
- Added `testAccentInsensitiveSearch` to `TestEntrySearcher` covering:
  - ASCII query matching accented entry and vice versa
  - Field-specific search with accent folding (title:, attachment:, tag:, group:)
  - Exact match (+) forces accent-sensitive, exclude (!) works with folding
  - Case-sensitive combined with accent-insensitive
- All existing tests pass unchanged
- Manual testing with KeePassXC GUI

## Type of change
- ✅ New feature (change that adds functionality)

### CODE REVIEW

**Comment by yyyyyyyan** [comment]
A couple of things I'd like to get input on:

1. 60f62592 fixes a bug where `EntrySearcher` was always constructed with hardcoded defaults, ignoring the user's saved config. I believe this went undetected because all existing toggle defaults were `false`, matching the constructor defaults, but I bumped into it with the new `accentSensitive` param defaulting to `true`. Not sure if this should stay here or be split into a separate issue/PR.

2. I went with accent-sensitive as the default (preserving current behavior), but honestly as a Brazilian Portuguese speaker I'd prefer accent-insensitive as the default. I'd like to know if anyone has any inputs regarding this.

**Comment by droidmonkey** [comment]
I am a fan of making the default more "liberal" and actually remove the option entirely. We already have an "exact match" prefix (ie, `+t:acćent`). If the user wants exact matching, accents are respected, otherwise the default is no accent matching.

**Comment by yyyyyyyan** [comment]
@droidmonkey done!

**Comment by yyyyyyyan** [comment]
@droidmonkey is anything needed on my end? Let me know if it needs any adjustments!!

**Comment by droidmonkey** [comment]
Nope you are gtg, I just need to manually test this to confirm functionality

File: src/core/Tools.cpp [hunk 1]
```diff
@@ -454,6 +454,33 @@ namespace Tools
         return string.replace("&", "&&");
     }
 
+    QString stripDiacritics(const QString& str)
+    {
+        // Fast path: pure ASCII has no diacritics to strip
+        bool ascii = true;
```
Comment (varjolintu) [inline src/core/Tools.cpp]: I think we already have `Tools::isAsciiString()`?

**Comment by mjuanes** [comment]
Any update on this?

File: src/core/Tools.cpp (hunk 1 above)
Comment (yyyyyyyan) [reply src/core/Tools.cpp]: @varjolintu you are right, I missed that, thank you!

Just one note + question: `stripDiactritics()` runs on the search hot path (per field x per entry x per term, every keystroke), and `isAsciiString()` currently (main) compiles a `QRegularExpression` on each call. To keep the fast-path allocation-free I modified `isAsciiString()` to be a plain scan (which was what my semi-duplicate code was doing):
```cpp
bool isAsciiString(const QString& str)
{
    for (const auto& ch : str) {
        if (ch.unicode() > 127) {
            return false;
        }
    }
    return true;
}
```
The only behavior change this does is that `isAsciiString("")` will return true instead of false, but I checked the callers and the impact seems to be null - it's unreachable at `PasskeyUtils::isDomain` because there it already short-circuits on `!domain.isEmpty()` first. This empty string behavior is also not covered by any test, so I believe it wasn't expected behavor any way.
Is that an ok addition for this PR or would you prefer not?

cc @droidmonkey who was keeping up with this pr

Comment (droidmonkey) [reply src/core/Tools.cpp]: Technically an empty string is an ascii string! Ok by my books
